core/consensus/grandpa/impl/voting_round_impl.hpp

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_CORE_CONSENSUS_GRANDPA_IMPL_VOTING_ROUND_IMPL_HPP
+#define KAGOME_CORE_CONSENSUS_GRANDPA_IMPL_VOTING_ROUND_IMPL_HPP
+
+#include "consensus/grandpa/voting_round.hpp"
+
+#include <boost/asio/basic_waitable_timer.hpp>
+#include <boost/signals2.hpp>
+#include "common/logger.hpp"
+#include "consensus/grandpa/environment.hpp"
+#include "consensus/grandpa/grandpa_config.hpp"
+#include "consensus/grandpa/vote_crypto_provider.hpp"
+#include "consensus/grandpa/vote_graph.hpp"
+#include "consensus/grandpa/vote_tracker.hpp"
+
+namespace kagome::consensus::grandpa {
+
+  class VotingRoundImpl : public VotingRound {
```

Comment:
I think we're gonna need to think of refactoring this class, it's monstrous.

Reply:
true, this is not the first version of it. It was much bigger before environment and vote crypto provider were introduced